File: server_nodejs/src/models/post.js
module.exports = (sequelize, Sequelize) => {
    class Post extends Sequelize.Model {}

    Post.init(
        {
            uid: {
                type: Sequelize.UUID,
                primaryKey: true,
            },
            title: {
                type: Sequelize.STRING,
            },
            content: {
                type: Sequelize.TEXT('long'),
                allowNull: false,
            },
            depth: {
                type: Sequelize.INTEGER,
                allowNull: false,
            },
            key_words: {
                type: Sequelize.STRING,
            },
            is_edited: {
                type: Sequelize.TINYINT,
                allowNull: false,
                defaultValue: 0,
            },
        },
        {
            sequelize,
            modelName: 'post',
            timestamps: true,
            paranoid: false,
            underscored: true,
        },
    );

    Post.associate = (models) => {
        Post.belongsTo(models.client_account, {
            foreignKey: 'created_by',
        });
        Post.belongsTo(Post, {
            foreignKey: {
                name: 'parent_post',
                allowNull: true,
            },
        });
        Post.hasMany(Post, {
            foreignKey: 'parent_post',
            as: 'children_posts',
        });
        Post.hasMany(models.vote, {
            foreignKey: 'postid',
            as: 'vote',
        });
    };

    return Post;
};

File: server_asp/src/models/Post.cs
using System;
using System.Collections.Generic;

namespace shieldtify.models;

public partial class Post
{
    public Guid Uid { get; set; }

    public string? Title { get; set; }

    public string Content { get; set; } = null!;

    public int Depth { get; set; }

    public string? KeyWords { get; set; }

    public sbyte IsEdited { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public Guid? CreatedBy { get; set; }

    public Guid? ParentPost { get; set; }

    public virtual ClientAccount? CreatedByNavigation { get; set; }

    public virtual ICollection<Post> ChildrenPosts { get; set; } = new List<Post>();

    public virtual ICollection<Vote> Votes { get; set; } = new List<Vote>();
}
